doc/modules/grid_search.rst

```diff
@@ -1,26 +1,27 @@
-.. currentmodule:: sklearn.grid_search
+
+
+.. currentmodule:: sklearn.model_selection
 
 .. _grid_search:
 
-===============================================
-Grid Search: Searching for estimator parameters
-===============================================
+===========================================
+Tuning the hyper-parameters of an estimator
+===========================================
 
-Parameters that are not directly learnt within estimators can be set by
-searching a parameter space for the best :ref:`cross_validation` score.
-Typical examples include ``C``, ``kernel`` and ``gamma`` for Support Vector
-Classifier, ``alpha`` for Lasso, etc.
+Hyper-parameters are parameters that are not directly learnt within estimators.
+In scikit-learn they are passed as arguments to the constructor of the
+estimator classes. Typical examples include ``C``, ``kernel`` and ``gamma``
+for Support Vector Classifier, ``alpha`` for Lasso, etc.
+
+It is possible and recommended to search the hyper-parameter space for the
+best :ref:`cross_validation` score.
 
 Any parameter provided when constructing an estimator may be optimized in this
-manner.  Specifically, to find the names and current values for all parameters
+manner. Specifically, to find the names and current values for all parameters
```

Comment:
I agree, here is a suggestion:

Note that it is common that a small subset of those parameters can have a large impact on the  predictive or computation performance of the model while others can be left to their default values. It is recommend to read the docstring of the estimator class to get a finer understanding of their expected behavior, possibly by reading the enclosed reference to the literature.